src/util.jl

```diff
@@ -19,6 +19,8 @@ TupleN{T,N} = NTuple{N,T}
 end
 
 # Base gives up on tuples for promote_eltype... (TODO can we improve Base?)
+_NTuple{T} = Tuple{T,Vararg{T}}
```

Comment:
The difference between `StaticArrays._NTuple` and `Base.NTuple` is not clear from their names. How about `_TupleOf` instead?
The name `_TupleOf` seems appropriate for the following behavior.
```julia
julia> () isa NTuple{N, Int} where N
true

julia> (1,) isa NTuple{N, Int} where N
true

julia> () isa StaticArrays._NTuple{Int}
false

julia> (1,) isa StaticArrays._NTuple{Int}
true
```

Reply:
Make sense. Thanks for the suggestion.